cardano-crypto-tests/src/Test/Crypto/DSIGN.hs

```diff
@@ -293,11 +360,216 @@ testDSIGNAlgorithm genSig genMsg name = adjustOption testEnough . testGroup name
     testEnough :: QuickCheckTests -> QuickCheckTests
     testEnough = max 10_000
 
+testDSIGNMAlgorithm
+  :: forall v. ( -- change back to DSIGNMAlgorithm when unsound API is phased out
+                 UnsoundDSIGNMAlgorithm IO v
+               , ToCBOR (VerKeyDSIGNM v)
+               , FromCBOR (VerKeyDSIGNM v)
+               -- DSIGNM cannot satisfy To/FromCBOR, because those
+               -- typeclasses assume that a non-monadic
+               -- encoding/decoding exists.  Hence, we only test direct
+               -- encoding/decoding for 'SignKeyDSIGNM'.
+               -- , ToCBOR (SignKeyDSIGNM v)
+               -- , FromCBOR (SignKeyDSIGNM v)
```

Comment:
I personally don't think we need this comment,  since it will be clear from `UnsoundDSIGNMAlgorithm` constraint. However feel free to keep it if beleive otherwise
```suggestion
```

Reply:
I don't think it's immediately obvious from the UnsoundDSIGNMAlgorithm constraint alone - if anything, an unsuspecting reader might wonder why CBOR encoding is not tested even with the unsound API explicitly allowed via the constraint.

Reply:
:+1: